PB-IDE-Tools/displayCcode.pb

```diff
@@ -0,0 +1,340 @@
+
+; -----------------------------------------------------------------------------
+
+; MIT License
+;
+; Copyright (c) 2018-2020 Sicro
+;
+; Permission is hereby granted, free of charge, to any person obtaining a copy
+; of this software and associated documentation files (the "Software"), to deal
+; in the Software without restriction, including without limitation the rights
+; to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+; copies of the Software, and to permit persons to whom the Software is
+; furnished to do so, subject to the following conditions:
+;
+; The above copyright notice and this permission notice shall be included in all
+; copies or substantial portions of the Software.
+;
+; THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+; IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+; FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+; AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+; LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+; OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+; SOFTWARE.
+
+; Tool Settings:
+; - Arguments: "%FILE" "%TEMPFILE"
+; - Event:     Menu Or Shortcut
+; For MacOS, the field "Commandline" must contain the full path to the executable
+; file, e.g.: .../Program.app/Contents/MacOS/Program
+
+EnableExplicit
+
+; =============
+;-Include Codes
+; =============
+
+Procedure.i OpenWithStandardProgram(FilePath$)
+  
+  Protected Result
+  
+  ; Avoid problems with paths containing spaces
+  FilePath$ = #DQUOTE$ + FilePath$ + #DQUOTE$
+  
+  CompilerSelect #PB_Compiler_OS
+    CompilerCase #PB_OS_Windows
+      ; https://docs.microsoft.com/en-us/windows/desktop/api/shellapi/nf-shellapi-shellexecutew
+      Result = Bool(ShellExecute_(0, "open", FilePath$, #Null, #Null, #SW_SHOW) > 32)
+    CompilerCase #PB_OS_Linux
+      ; https://portland.freedesktop.org/doc/xdg-open.html
+      Result = Bool(RunProgram("xdg-open", FilePath$, GetCurrentDirectory()))
+    CompilerCase #PB_OS_MacOS
+      Result = Bool(RunProgram("open", FilePath$, GetCurrentDirectory()))
+  CompilerEndSelect
+  
+  ProcedureReturn Result
+  
+EndProcedure
+
+;XIncludeFile "../File/GetFileContentAsString.pbi"
+
+; ================
+;-Define Constants
+; ================
+
+#Program_Name     = "Display C-Code"
+#ErrorWindowTitle = #Program_Name + " - ERROR"
+
+#Window_Main = 0
+
+Enumeration Gadget
+  #Editor_Output
+  #Button_OpenStandardEditor
+  #Button_CopyToClipboard
+EndEnumeration
+
+; ======================
+;-Define Local Variables
+; ======================
+
+Define compilerHomePath$, compilerFilePath$, workingDirectoryPath$, codeFilePath$, codeTempFilePath$,
+       asmCodeFilePath$, exeFilePath$, outputFilePathForStandardProgram$, compilerParameters$,
+       compilerUserParameters$
+Define asmCode$, output$, compilerOutput$
```

Comment:
Please change:
- `asmCodeFilePath$` to `cCodeFilePath$`
- `asmCode$` to `asmCode$`